java/PrimaDla/src/org/primaresearch/dla/page/io/xml/XmlPageReader.java

```diff
@@ -143,8 +143,14 @@ public Page read(InputSource source) throws UnsupportedFormatVersionException {
 		
 		Page page = null;
 		
-		if (!lastErrors.hasErrors())
-			page = pageHandler.getPageObject();
+		if (lastErrors.hasErrors()) {
+			StringBuilder sb = new StringBuilder();
+			for (IOError error : getErrors())
+				sb.append(error.getMessage());
+			throw new UnsupportedFormatVersionException(sb.toString());
```

Comment:
The new code will not only report the error, but also throw an error instead of returning null. Can that cause new issues?

Reply:
Since this is a _library_, yes, throwing an exception is the design of choice. Downstream applications can decide how to act on that. So the viewer or the converter might want to present an error message about invalid input. But for the moment, the stack trace will suffice – remember the user had _nothing_ up until now. (I remember hours-long hunts on failed PAGE files for sublime non-syntactic errors like missing `@conf` or dangling `@regionRef`.)